### ISSUE TITLE
Web accessibility - Search form control does not have a corresponding label.

### ISSUE THREAD

**Issue by natnael2012** [learningequality/ka-lite#3299]
Branch - develop

Expected behavior - The search form control should have a corresponding label. If a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users. Form labels also provide visible descriptions and larger clickable targets for form controls.

Standards and Guidelines
•	Section 508 (n) http://webaim.org/standards/508/checklist#standardn
•	1.1.1 Non-text Content (Level A) http://webaim.org/standards/wcag/checklist#sc1.1.1
•	1.3.1 Info and Relationships (Level A) http://webaim.org/standards/wcag/checklist#sc1.3.1
•	2.4.6 Headings and Labels (Level AA) http://webaim.org/standards/wcag/checklist#sc2.4.6
•	3.3.2 Labels or Instructions (Level A) http://webaim.org/standards/wcag/checklist#sc3.3.2

Current behavior – The search form control does not have a corresponding label.

Snapshot of the display inspected by Wave checker
![label](https://cloud.githubusercontent.com/assets/8178368/6651152/fe4b4014-c9f2-11e4-95dc-ae6cc390d583.png)

**Comment by radinamatic** [comment]
I'm working on this one and will present a PR today. @natnael2012 do you have a PR proposal?

Is there a way one can assign/reserve the issue so we do not have two people working on the same issue? There is plenty to go around and fix... ;)

**Comment by natnael2012** [comment]
I didn't know you were working on this issue. And yes, I have a PR proposal.

**Comment by radinamatic** [comment]
You are right, I should have posted the issue first, but I mentioned it in the chat yesterday with @mike and I haven't seen any issues/PRs about it... Ok, you go for this one, since you "beat" me by posting the issue first... :)
   
How were you planning on approaching it? Develop branch doesn't have the "Search" text to use as a label (master had a text button that could have been labeled). Did you plan on adding the text back in front of the input control?

**Comment by natnael2012** [comment]
:) Their is small space available so I added a label in front of the searchbox

**Comment by radinamatic** [comment]
Yeah, that's one option... :) 
Looking forward to seeing the PR.

**Comment by natnael2012** [comment]
:-) , well i have updated my previous PR  by solving this issue. You cant request for new PR at the same branch if you already have open one.

**Comment by natnael2012** [comment]
:-) , well i have updated my previous PR
by solving this issue. You cant request
for new PR at the same branch if you
already have one open.

**Comment by radinamatic** [comment]
Nicely done!
However, instead of using the **label** (WCAG H44), which forced you to style and reintroduce the text I assume was removed to save space in the first place, I would recommend the WCAG2.0 technique H65 and use the **title** attribute instead. This way you don't need the text, visually the UI remains the same, and you are still respecting the guidelines.

So, instead of:
```html
<p class="search-label search-label-mini"><label for="query">Search</label></p>
<input type="text" name="query" id="search_box" class="form-control pull-left" placeholder="{% trans 'Topic, video, exercise...' %}" />
```
I would recommend:
```html
<input type="text" name="query" id="search_box" class="form-control pull-left" title="Type search term here" placeholder="{% trans 'Topic, video, exercise...' %}" />
<button class="btn btn-default" id="search_btn" type="submit" value="Search"><i class="glyphicon glyphicon-search"></i></button>
```
WAVE checker tool will not report this solution as an error (just a warning).
I've updated my previous PR (#3292) with this change.

**Comment by MCGallaspy** [comment]
@radinamatic has this been resolved?

**Comment by radinamatic** [comment]
Yep, close away!
